### PR TITLE
PR: forpro-vis/student-view-rework

### PR DESCRIPTION
**Pull request by TJHexler**
# Description of changes

new student course overview navigation

## Checklist for reviewer

- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code follows the [coding guidelines](https://github.com/MEITREX/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

File: components/ChapterOverviewItem.tsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+                strokeLinecap: "round",
+              },
+              color: selected ? "#F27900" : "#84BFE6",
+            }}
+          />
+        )}
+        {(disabled && (
+          <LockOutlined className="absolute w-9 h-9 text-gray-300" />
+        )) ||
+          (progress >= 0 && progress < 100 && (
```
Comment (becknik) [inline components/ChapterOverviewItem.tsx:79]: Why `progress >= 0`?

File: components/ChapterOverviewItem.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            <div
+              className="absolute text-sm font-bold"
+              style={{ color: selected ? "#F27900" : "#84BFE6" }}
```
Comment (becknik) [inline components/ChapterOverviewItem.tsx:82]: Use `theme => theme.palette.success` etc instead of hex codes; refactor the codes into constants

See: https://mui.com/material-ui/customization/palette/#default-colors

File: components/ChapterOverviewItem.tsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+              className="absolute w-10 h-10"
+              style={{ color: selected ? "#F27900" : "#84BFE6" }}
+            />
+          ))}
+      </div>
+      <div
+        onClick={!disabled ? onClick : undefined}
+        className="absolute top-full mt-2 flex flex-col items-center w-40 h-auto"
+      >
+        {/* Dreieck */}
```
Comment (becknik) [inline components/ChapterOverviewItem.tsx:98]:
```suggestion
        {/* Triangle */}
```

File: components/ChapterOverviewItem.tsx [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        <div className="w-0 h-0 border-l-8 border-r-8 border-b-8 border-transparent border-b-white" />
+
+        {/* Tooltip-Blase */}
```
Comment (becknik) [inline components/ChapterOverviewItem.tsx:101]:
```suggestion
        {/* Tooltip-Buble */}
```

File: package.json [hunk 5]
```diff
@@ -50,6 +50,7 @@
     "oidc-client-ts": "^2.4.0",
     "pdfobject": "^2.3.0",
     "postcss": "8.4.23",
+    "prop-types": "^15.8.1",
```
Comment (becknik) [inline package.json:53]: doesn't seem to be used anywhere in this PR

File: components/ChapterOverviewItem.tsx [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+  _chapter: ChapterOverviewItemFragment$key;
+  selected: boolean;
+  onClick: () => void;
+}) {
+  const chapter = useFragment(ChapterFragment, _chapter);
+  const progress = Math.round(chapter.userProgress.progress);
+  const suggestedStartDate = Date.parse(
+    chapter.suggestedStartDate ?? chapter.startDate
+  );
+  const disabled = suggestedStartDate.valueOf() > Date.now();
```
[the remark this answers is not in the document]
Comment (sangaran10) [reply components/ChapterOverviewItem.tsx:35]: I tried and IDE shows error for your suggestions, because suggestedStartDate is type number and so is Date.now(), so it doesn't work with suggestedStartDate > new Date() and you can't use suggestedStartDate.getTime()

File: components/ChapterOverviewItem.tsx (hunk 1 above)
Comment (sangaran10) [reply components/ChapterOverviewItem.tsx:79]: isnt it safer to check that it is between 0 and 100

Comment (TJHexler) [reply components/ChapterOverviewItem.tsx]: Removed progress >= 0 :) No need for this check.

File: components/ChapterOverviewItem.tsx (hunk 2 above)
Comment (TJHexler) [reply components/ChapterOverviewItem.tsx]: swapped hexcodes to themes 👍
